packages/api/src/webhooks/cannon.ts

```diff
@@ -222,6 +239,69 @@ export default class WebhookCannon {
     );
   }
 
+  async notifyDisabledWebhook(
+    trigger: messages.WebhookTrigger,
+    params?: any,
+    err?: any
+  ) {
+    try {
+      if (trigger.user.emailValid) {
+        if (
+          !(this.supportAddr && this.sendgridTemplateId && this.sendgridApiKey)
+        ) {
+          console.error(
+            `Webhook Cannon| Unable to send notification email to user, id: ${trigger.id}, streamId: ${trigger.stream?.id}`
+          );
+          console.error(
+            `Sending emails requires supportAddr, sendgridTemplateId, and sendgridApiKey`
+          );
+
+          return;
+        }
+
+        let signature_header = "";
+        if (trigger.webhook.sharedSecret) {
+          signature_header = "-H " + params.headers["Livepeer-Signature"];
+        }
+
+        let payload = params.body;
+
+        await sendgridEmail({
+          email: trigger.user.email,
+          supportAddr: this.supportAddr,
+          sendgridTemplateId: this.sendgridTemplateId,
+          sendgridApiKey: this.sendgridApiKey,
+          subject: "Your webhook has been disabled",
+          preheader: "Failure notification",
+          buttonText: "Manage your webhooks",
+          buttonUrl: `https://${this.frontendDomain}/dashboard/developers/webhooks`,
+          unsubscribe: `https://${this.frontendDomain}/contact`,
+          text: [
+            `Your webhook failed to receive this payload in the last 24 hours: `,
+            `<code>${payload}</code>`,
+            `This is the error we are receiving:`,
+            `${err}`,
+            `We disabled your webhook, please check your configuration and try again.`,
+            `If you want to try yourself the call we are making, here is a curl command for that:`,
+            `<code>curl -X POST -H "Content-Type: application/json" -H "user-agent: livepeer.com" ${signature_header} -d '${payload}' ${trigger.webhook.url}</code>`,
+          ].join("\n\n"),
+        });
+      } else {
+        console.error(
+          `Webhook Cannon| User email is not valid, id: ${trigger.id}, streamId: ${trigger.stream?.id}`
+        );
+        return;
+      }
+      console.log(
+        `Webhook Cannon| Email sent to user, id: ${trigger.id}, streamId: ${trigger.stream?.id}`
+      );
+    } catch (err) {
```

Comment:
WDYT of moving the try/catch to the calling function just so we don't have the huge block here? 🥴